lib/components/result-view/index.js

```diff
@@ -0,0 +1,107 @@
+/* @flow */
+
+import { CompositeDisposable, Point } from "atom";
+import React from "react";
+import { action, observable } from "mobx";
+import { createImmutableOutput } from "@nteract/commutable";
+
+import { reactFactory } from "./../../utils";
+import ResultViewComponent from "./result-view";
+
+import type { IObservableArray, IObservableValue } from "mobx";
+
+import type { ImmutableOutput } from "@nteract/commutable/lib/types";
+
+import type { Output, StreamOutput } from "@nteract/commutable/lib/v4";
+
+const outputTypes = ["execute_result", "display_data", "stream", "error"];
+
+/**
+ * https://github.com/nteract/hydrogen/issues/466#issuecomment-274822937
+ * An output can be a stream of data that does not arrive at a single time. This
+ * function handles the different types of outputs and accumulates the data
+ * into a reduced output.
+ *
+ * @param {Array<Object>} outputs - Kernel output messages
+ * @param {Object} output - Outputted to be reduced into list of outputs
+ * @return {Array<Object>} updated-outputs - Outputs + Output
+ */
+export function reduceOutputs(
+  outputs: IObservableArray<ImmutableOutput>,
+  output: Output
+): IObservableArray<any> {
+  const last = outputs.length - 1;
+  if (
+    outputs.length > 0 &&
+    output.output_type === "stream" &&
+    outputs[last].get("output_type") === "stream"
+  ) {
+    const stream: StreamOutput = output;
+    outputs[last] = outputs[last].update("text", text => text + stream.text);
+    return outputs;
```

Comment:
Since we're in the midst of this -- this is exactly the segment I wanted to handle carriage returns and other terminal movement commands in -- then it's only done once and not on every render.